revenue_app/views.py

```diff
@@ -146,39 +154,101 @@
 ]
 
 
-class Dashboard(TemplateView):
+class QueriesRequiredMixin():
+    def dispatch(self, request, *args, **kwargs):
+        if any([
+            request.session.get(dataframe) is None
+            for dataframe in ['transactions', 'corrections', 'organizer_sales']
+        ]):
+            return HttpResponseRedirect(resolve_url('make-query'))
+        return super(QueriesRequiredMixin, self).dispatch(request, *args, **kwargs)
```

Comment:
Since this is  Python 3 you should be able to just write `super().dispatch(request, *args, **kwargs)`